### ISSUE TITLE
Upgrade to Node v14

### ISSUE THREAD

**Issue by dylburger** [PipedreamHQ/pipedream#164]

**Comment by chrigi** [comment]
Is there any plan in 2021 to upgrade the node runtime? Maybe even to v14?

**Comment by dylburger** [comment]
@chrigi is there a specific use case you're blocked on, or a feature you'd like to use from Node 12 / 14? Any detail on the use case is helpful for us.

**Comment by chrigi** [comment]
@dylburger I was trying to use [Array.prototype.flat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) which is not available and I am missing the nullish coalescing and optional chaining operators as I got quite used to them already. So for me it's mainly about ES language support, not directly missing Node APIs.

NodeJS v10 will also reach the end of it's maintenance period in April 2021 so I was interested if it's still the goal to upgrade to v12 or jump to v14 LTS directly. v14 would bring all ES features that I'm missing while v12 would only bring a few.

**Comment by dylburger** [comment]
Very much appreciate the detail @chrigi . I'll update this ticket as soon as we address it.

**Comment by DanielsLuz** [comment]
I'm using the optional chaning on a workflow, however, now I'm trying to extract a step into it's own action, and the `pd publish` command fails when it contains a `?.` operator. When removed, it publishes correctly.

Minimal action:
```js
module.exports = {
  name: "Optional chaining",
  description: "Optional chaining demo",
  key: "optional_chaining_demo",
  version: "0.0.1",
  type: "action",
  props: {},
  async run (event) {
    return event?.foo
  },
}
```

The response is:
```
component create api call responded with status: 400, body: {"error":"{\"name\":\"UserError\",\"message\":\"SyntaxError: Unexpected token '.'\"}"}
```

**Comment by dylburger** [comment]
@DanielsLuz yes thank you, this should be fixed soon when we ship #1713
